python/ray/dag/compiled_dag_node.py

```diff
@@ -2240,6 +2240,327 @@ async def execute_async(
         self._execution_index += 1
         return fut
 
+    def visualize_ascii(self):
+        """
+        Visualize the compiled graph in
+        ASCII format with directional markers.
+
+        This function generates an ASCII representation of a CG,
+        where each task node is labeled,
+        and edges use `<` and `>` markers to show data flow direction.
+
+        High-Level Algorithm:
+        - Topological Sorting: Sort nodes topologically to organize
+            them into layers based on dependencies,
+            ensuring nodes appear in a logical order.
+        - Grid Initialization: Set up a 2D grid canvas with dimensions based
+            on the number of layers and the maximum number of nodes per layer.
+        - Node Placement: Position each node on the grid according to its
+            layer and relative position within that layer.
+            Spacing is added for readability, and directional markers (`<` and `>`)
+            are added to edges to show input/output flow clearly.
+
+        This method should be called
+          **after** compiling the graph with `experimental_compile()`.
+
+        Returns:
+            ASCII representation of the CG with Nodes Information,
+            Edges Information and Experimental Graph Built.
```

Comment:
What is "Experimental Graph"?

Reply:
removed the Experimental  as originally I think this is just an Experimental that may have better solutions..while its not